kythe/typescript/indexer.ts

```diff
@@ -1139,15 +1139,39 @@ class Visitor {
           const kParent = this.getSymbolName(parentSym, TSNamespace.TYPE);
           this.emitEdge(kFunc, 'childof', kParent);
         }
-      }
 
-      // TODO: emit an "overrides" edge if this method overrides.
-      // It appears the TS API doesn't make finding that information easy,
-      // perhaps because it mostly cares about whether types are structrually
-      // compatible.  But I think you can start from the parent class/iface,
-      // then from there follow the "implements"/"extends" chain to other
-      // classes/ifaces, and then from there look for methods with matching
-      // names.
+        // Emit "overrides" edges if this method overrides extended classes or
+        // implemented interfaces, which are listed in the Heritage Clauses of
+        // a class or interface.
+        //     class X extends A implements B, C {}
+        //             ^^^^^^^^^-^^^^^^^^^^^^^^^----- `HeritageClause`s
+        // Look at each type listed in the heritage clauses and traverse its
+        // members. If the type has a member that matches the method visited in
+        // this function (`kFunc`), emit an "overrides" edge to that member.
+        if (funcSym && decl.parent.heritageClauses) {
+          for (const heritage of decl.parent.heritageClauses) {
+            for (const baseType of heritage.types) {
+              const baseSym = this.getSymbolAtLocation(baseType.expression);
+              if (!baseSym) {
+                this.todo(baseType, `heritage type ${baseSym} has no symbol`);
+                continue;
+              }
+              const funcName = funcSym.name;
+              const funcFlags = funcSym.flags;
+              const overriden =
+                  toArray((baseSym.members || new Map()).values())
```

Comment:
I read this like four times before I followed.

This might be easier to follow if the missing members thing was handled first, like:

```
baseSym.members ? toArray(...).find(...) : false;
```

or with an if statement.